core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -919,6 +919,7 @@ public void sendRequest(final DiscoveryNode node, final long requestId, final St
                     transportServiceAdapter.onRequestSent(node, requestId, action, request, finalOptions);
                 }
             };
+            Channel targetChannel = nodeChannel(node, options);
```

Comment:
as I said, I think doing it early to catch node close quickly it's intentional (though not a big deal). Can we keep it as is then?